src/main/java/com/team701/buddymatcher/controllers/communication/CommunicationController.java

```diff
@@ -89,6 +90,33 @@ public CommunicationController(CommunicationService communicationService, UserSe
         }
     }
 
+
+    @Operation(summary = "Delete method to delete a user's conversation")
+    @DeleteMapping(path = "/messages/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Void> deleteUserMessages(@Parameter(hidden = true) 
+                                                   @SessionAttribute("UserId") Long userId, 
+                                                   @PathVariable("id") Long buddyId) {
+        try {
+            // Check if the targeted user is valid/a buddy
+            List<User> buddies = userService.retrieveBuddiesByUserId(userId);
+            boolean isValid = false;
+            for (int i = 0; i < buddies.size(); i++) {
+                if (buddies.get(i).getId() == buddyId) {
+                    isValid = true;
+                    break;
+                }
+            }
+            if (isValid) {
+                communicationService.deleteMessagesBetweenUsers(userId, buddyId);
+                return new ResponseEntity<>(HttpStatus.OK);
+            } else {
+                throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Buddy not found");
```

Comment:
This is redundant. It should be thrown when there is no user in the database. There is no requirement for checking if the two users are buddies. A suggsted solution is to add `throw NoSuchElementException` to the ``deleteMessagesBetweenUsers()`.

Reply:
I tried adding the throw NoSuchElementException to deleteMessagesBetweenUsers() like you suggested, but that didn't work. I've tried just checking if the user exist by trying to retrieve it instead and that seems to work.